docs/source/getting-started/teacher_draft.ipynb

```diff
@@ -0,0 +1,185 @@
+    "privacy_loss = dp.loss_of(epsilon=1.)\n",
+    "privacy_measure, d_out = privacy_loss\n",
+    "\n",
+    "## step 3 / nothing changed\n",
+    "col_names = [\n",
+    "   \"name\", \"sex\", \"age\", \"maritalStatus\", \"hasChildren\", \"highestEducationLevel\", \n",
+    "   \"sourceOfStress\", \"smoker\", \"optimism\", \"lifeSatisfaction\", \"selfEsteem\"\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## the way to read data in changed\n",
+    "#scan csv with the url wasn't working directly so it wasn't able to be read in \n",
+    "#directly as a lazy frame; however if we could debug this that would be great\n",
+    "#prehaps move this import up but kept it here to keep same style as og page \n",
+    "import polars as pl\n",
+    "data_url = \"https://raw.githubusercontent.com/opendp/opendp/sydney/teacher_survey.csv\"\n",
+    "data = pl.read_csv(data_url)\n",
+    "data.columns = col_names\n",
+    "data = pl.LazyFrame(data)\n",
+    "\n",
+    "#cast all integers to floats since fill nan can only work on floats\n",
+    "data = data.cast({pl.Int64: pl.Float64})\n",
+    "\n",
+    "context = dp.Context.compositor(\n",
+    "    data=data,\n",
+    "    privacy_unit=privacy_unit,\n",
+    "    privacy_loss=privacy_loss,\n",
+    "    split_evenly_over=3, \n",
+    "    margins={\n",
+    "        (\"age\", ): dp.Margin(public_info=\"lengths\", max_partition_length=8000, max_num_partitions=1),\n",
+    "        (): dp.Margin(public_info=\"lengths\", max_partition_length=8000, max_num_partitions=1),\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We set the `max_partition_length` to be 8000 since we know our data has less than 8000 entries. The `max_num_partitions` is set to 1 since the order that our methods are using on the backgrounds requires `max_num_partitions` to be specified and it's 1 since one of our queries require only 1 partition. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Based on previous examples, one of the initial ways you may try to find the count of age directly could be \n",
+    "\n",
+    "```\n",
+    "context.query().select(pl.col(\"age\").len().dp.noise()\n",
+    ").release().collect() \n",
+    "```\n",
+    "\n",
+    "However, this method won't work yet since in this case we're using `expr.len`, which isn't supported yet by OpenDP and is different than `pl.len` in polars. Therefore, we'll directly get the length of the dataframe by doing `pl.len().dp.noise()`. However, we also need to impose some type of bound, which we can do by filtering the variable of interest. "
```

Comment:
this seems like TMI. why are we referencing unsupported expressions in a beginner tutorial?